src/components/metadataFormControls/AggregationLevelMultiSelect/AggregationLevelMultiSelect.tsx

```diff
@@ -0,0 +1,97 @@
+import i18n from '@dhis2/d2-i18n'
+import { MultiSelect, MultiSelectOption } from '@dhis2/ui'
+import React, { forwardRef, useImperativeHandle } from 'react'
+import classes from './AggregationLevelMultiSelect.module.css'
+import { useOptionsQuery } from './useOptionsQuery'
+
+function Error({
+    msg,
+    onRetryClick,
+}: {
+    msg: string
+    onRetryClick: () => void
+}) {
+    return (
+        <div className={classes.error}>
+            <div className={classes.errorInnerWrapper}>
+                <span className={classes.loadingErrorLabel}>{msg}</span>
+                <button
+                    className={classes.errorRetryButton}
+                    type="button"
+                    onClick={onRetryClick}
+                >
+                    {i18n.t('Retry')}
+                </button>
+            </div>
+        </div>
+    )
+}
+
+interface AggregationLevelMultiSelectProps {
+    onChange: ({ selected }: { selected: string[] }) => void
+    onRetryClick: () => void
+    inputWidth?: string
+    placeholder?: string
+    selected?: string[]
+    showAllOption?: boolean
+    onBlur?: () => void
+    onFocus?: () => void
+}
+
+export const AggregationLevelMultiSelect = forwardRef(
+    function AggregationLevelMultiSelect(
+        {
+            onChange,
+            inputWidth,
+            selected,
+            showAllOption,
+            placeholder = i18n.t('Aggregation level(s)'),
+            onBlur,
+            onFocus,
+            onRetryClick,
+        }: AggregationLevelMultiSelectProps,
+        ref
+    ) {
+        const optionsQuery = useOptionsQuery()
+        const { refetch } = optionsQuery
+
+        useImperativeHandle(ref, () => ({ refetch }), [refetch])
```

Comment:
Could we have a prop and render the button inside this component instead?

Reply:
Yes, I was thinking that we might use this as a filter in the filter section as well where we wouldn't use the buttons